### ISSUE TITLE
esModuleInterop doesn't affect reexports

### ISSUE THREAD

**Issue by wojpawlik** [microsoft/TypeScript#39464]
**TypeScript Version:**  4.0.0-dev.20200707

**Search Terms:**

esmoduleinterop reexport

**Code**

https://gist.github.com/b48d1791d83ff67581238190eb583f67

**Expected behavior:**

`lib/index.js` exports `require('./answer')` (`42`)

**Actual behavior:**

`lib/index.js` exports `require('./answer').default` (`undefined`)

**Comment by rbuckton** [comment]
```ts repro
// @showEmit
// @showEmittedFile: b.js
// @module: commonjs
// @allowJs: true
// @esModuleInterop: true
// @strict: true
// @filename: a.js
module.exports = 42;
// @filename: b.ts
export {default} from './a'

```

[Workbench Repro](https://www.typescriptlang.org/dev/bug-workbench/?target=1&ts=4.0.5#code/PTAEAEGcAsHsHcCiBbAlgFwLACgQRgihugKYAmAYqgDYkBcoARgHQBWkOe4ysZArrQYBjWMh4A7dpzDgAhtWoIAUpAboATnxLSIJSAFleAkgElxpdbAAOazdtwzIG1EPS2tO8ADMaJcbOR6UFk2DmweflpmEgAPK1h1dEhQAF5QABYAJgBuTx9af0CGFiScWPjE0ABvMhIvWQF0AF9QL0tkUAByZmBZTpwgA)

**Comment by rbuckton** [comment]
This was fixed by #39490 and shipped in 4.0
